### DISCUSSION TITLE
How to Add a New Language?

### DISCUSSION THREAD

**Discussion by SyamsQ** [SWivid/F5-TTS#614] [Q&A]
I found a model for Indonesian. What should I do to run the model in F5-TTS?
![image](https://github.com/user-attachments/assets/6556fd38-ac7b-48b7-beb3-12378cff0655)
![image](https://github.com/user-attachments/assets/51856d0a-0829-4e54-939c-661cd77daa37)

**Comment by SWivid** [comment]
run local gradio
install this repo and run `f5-tts_infer-gradio` (see readme
click custom, filling like the way of given format

**Reply by SyamsQ** [reply]
Is this the Gradio you're referring to? How to run f5-tts_infer-gradio? 
![image](https://github.com/user-attachments/assets/fb38bb04-5011-4768-9e79-b9d4fbe8e195)

**Reply by SyamsQ** [reply]
I successfully executed the model from Huggingface. But some of the words in this audio are not pronounced correctly. How can I fix it?
![image](https://github.com/user-attachments/assets/bf9a7151-7a78-4ba6-a309-02221a6c3228)

**Reply by SWivid** [reply]
Hi @SyamsQ , you could continue to finetune the Indonesian model with more data to improve the pronunciation,
and might also have some discussion with the Indonesian model holder who has more expertise to the language.

**Reply by SyamsQ** [reply]
How to finetune the model with more data?

**Reply by SWivid** [reply]
could follow the readme for training

**Reply by SyamsQ** [reply]
readme what?
There is no training guide here.
![image](https://github.com/user-attachments/assets/c71117c1-7ea0-479f-98da-c72e014c1de6)

**Reply by SWivid** [reply]
the repo here

main readme: https://github.com/SWivid/F5-TTS/blob/main/README.md
training readme: https://github.com/SWivid/F5-TTS/tree/main/src/f5_tts/train
inference readme: https://github.com/SWivid/F5-TTS/blob/main/src/f5_tts/infer/README.md

**Reply by jingzhoucao05-lang** [reply]
Hello, can you generate Indonesian in f5tts now

**Reply by SyamsQ** [reply]
> Hello, can you generate Indonesian in f5tts now

how can?

**Reply by jingzhoucao05-lang** [reply]
Inputting a text can generate Indonesian using f5tts

**Reply by SyamsQ** [reply]
> I successfully executed the model from Huggingface. But some of the words in this audio are not pronounced correctly. How can I fix it?

I know that. But some of the words in this audio are not pronounced correctly. How can I fix it?

<img width="1109" height="809" alt="image" src="https://github.com/user-attachments/assets/f0012dfe-cfb8-4c76-9cc7-6e77abef2f10" />
